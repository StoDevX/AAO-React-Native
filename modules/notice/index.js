// @flow

export {LoadingView} from './loading'
export {NoticeView} from './notice'
