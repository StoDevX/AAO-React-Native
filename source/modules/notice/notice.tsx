import * as React from 'react'
import {
	ActivityIndicator,
	StyleProp,
	StyleSheet,
	Text,
	TextStyle,
	View,
	ViewStyle,
} from 'react-native'
import * as c from '../colors'
import {Button} from '../button'
import {Heading} from '../markdown'

const styles = StyleSheet.create({
	container: {
		flex: 1,
		paddingHorizontal: 30,
		justifyContent: 'center',
		alignItems: 'center',
		backgroundColor: c.systemBackground,
	},
	text: {
		textAlign: 'center',
		color: c.label,
	},
	spinner: {
		alignItems: 'center',
		justifyContent: 'center',
		padding: 8,
	},
})

interface Props {
	buttonDisabled?: boolean
	header?: string
	text?: string
	style?: StyleProp<ViewStyle>
	spinner?: boolean
	buttonText?: string
	onPress?: () => unknown
	textStyle?: StyleProp<TextStyle>
}

export function NoticeView(props: Props): React.JSX.Element {
	let {header, text, style, textStyle} = props
	let {buttonDisabled, buttonText, onPress} = props
	let {spinner} = props

	return (
		<View style={[styles.container, style]}>
			{spinner ? <ActivityIndicator style={styles.spinner} /> : null}

			{header ? (
				<Heading level={1} style={textStyle}>
					{header}
				</Heading>
			) : null}

			<Text selectable={true} style={[styles.text, textStyle]}>
				{text || 'Notice!'}
			</Text>

			{buttonText ? (
				<Button
					disabled={buttonDisabled}
					onPress={onPress}
					title={buttonText}
				/>
			) : null}
		</View>
	)
}
