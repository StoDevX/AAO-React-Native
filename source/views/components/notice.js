// @flow
import * as React from 'react'
import {ActivityIndicator, StyleSheet, Text, View} from 'react-native'
import * as c from './colors'
import {Button} from './button'
import {Heading} from './markdown/heading'
import {Viewport} from './viewport'

const styles = StyleSheet.create({
	container: {
		flex: 1,
		paddingHorizontal: 16,
		justifyContent: 'center',
		alignItems: 'center',
		backgroundColor: c.white,
	},
	text: {
		textAlign: 'center',
	},
	spinner: {
		alignItems: 'center',
		justifyContent: 'center',
		padding: 8,
	},
})

type Props = {
	buttonDisabled?: boolean,
	header?: string,
	text?: string,
	style?: any,
	spinner?: boolean,
	buttonText?: string,
	onPress?: () => any,
	textStyle?: any,
}

export function NoticeView(props: Props) {
	const {header, text, style, textStyle} = props
	const {buttonDisabled, buttonText, onPress} = props
	const {spinner} = props

	return (
		<Viewport
			render={() => (
				<View style={[styles.container, style]}>
					{spinner ? <ActivityIndicator style={styles.spinner} /> : null}

					{header ? (
						<Heading level={1} style={textStyle}>
							{header}
						</Heading>
					) : null}

					<Text selectable={true} style={[styles.text, textStyle]}>
						{text || 'Notice!'}
					</Text>

					{buttonText ? (
						<Button
							disabled={buttonDisabled}
							onPress={onPress}
							title={buttonText}
						/>
					) : null}
				</View>
			)}
		/>
	)
}
