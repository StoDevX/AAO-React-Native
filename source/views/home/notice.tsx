import React from 'react'
import {View, Text, StyleSheet} from 'react-native'
import * as c from '@frogpond/colors'
import sample from 'lodash/sample'
import {CELL_MARGIN} from './button'
import {isDevMode} from '@frogpond/constants'

let messages = [
	'☃️ An Unofficial App Project ☃️',
	'For students, by students',
	'By students, for students',
	'An unofficial St. Olaf app',
	'For Oles, by Oles',
	'☃️',
	'🦁',
	'Made with ❤️ in Northfield, MN',
]

if (isDevMode()) {
	messages = [
		...messages,
		'made with  ⃟ in Ñ̸̞͖̘̱̰̥͇̗̂͌̇̎͊ͯ̎̓̎ͥ̋̐ͤͪͭ̚͘͢͢ø̸̛̞͊̎ͩ̍̉̑ͯͫͥ̚͟ͅ ̱̬̹̱̦®̵̬͖͙̻̩͓̖̠͉͈͍̈́̅͂͛̅̀͗ͤ̓́͡†̵̧͙̥̫̫͎̘̩̲̥̖̈̌͋̀ͨ̑̽̍̆̓̒̒̄̈́͒̓̕͜ ͍̩̫̼ͅ˙̶͕̰̗͓̯̫̲̮͕̪̝͎̩̬̺̔ͯ̌̈̽̌ͨ͊͊͐̀͆̽̐̓̃́̚͢͟ ̞̞̤ƒ͚͙̤ͭͪ͑̄͆͑ͯ̆͗̆ͨ̍̀͟͢ ̙͎̝͕͔̠͉̩̯͕͚̗̤ͅî̹̗̩̫̝̝͙̠̹̣̺̤̆ͭ̾̋ͬ̂ͫ̃̏ͥͬ́͜͠é̚ ̸͔͕̗̞̰́̅̅͒ ̪̩̞̰̫͓̞̱̫̞̭̯¬ͫ̾̆ ̍ͣ̎̀ͫͪͪ̋͌̂ ̪̘̯̝̤͌̆ͮ̕͜͜͡∂̢̛͕̻͖̈͌ͮ̂̾ͪͪ̑͋͂̂̂̂̈́̈́̓̌̍̌͜͞ ͙̫̤',
		'made with ∆ in Ñø®†˙ƒîé¬∂',
		'Made with 🤞 in ⬆️🌾',
		'⬆️🌾=🐄🏫♥️',
	]
}

export function UnofficialAppNotice(): JSX.Element {
	return (
		<View style={[styles.wrapper, styles.background]}>
			<Text style={styles.text}>{sample(messages)}</Text>
		</View>
	)
}

const styles = StyleSheet.create({
	wrapper: {
		justifyContent: 'center',
		marginBottom: CELL_MARGIN,
		marginHorizontal: CELL_MARGIN,
	},
	background: {
		backgroundColor: c.secondarySystemFill,
		borderRadius: 7,
	},
	text: {
		color: c.secondaryLabel,
		padding: 8,
		textAlign: 'center',
	},
})
