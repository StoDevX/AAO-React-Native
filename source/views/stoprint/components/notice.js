// @flow

import React from 'react'
import {
	Platform,
	Text,
	ScrollView,
	StyleSheet,
	RefreshControl,
} from 'react-native'
import Icon from 'react-native-vector-icons/Ionicons'
import {NoticeView} from '@frogpond/notice'
import {sto} from '../../../lib/colors'
import {Timer} from '@frogpond/timer'

type Props = {
	buttonText: string,
	description?: string,
	header: string,
	onPress: () => any,
	refresh: () => any,
	text: string,
}

export class StoPrintNoticeView extends React.PureComponent<Props> {
	render() {
		const {buttonText, description, header, onPress, text} = this.props

		return (
			<Timer
				interval={5000}
				invoke={this.props.refresh}
				moment={false}
				render={({refresh, loading}) => (
					<ScrollView
						contentContainerStyle={styles.content}
						refreshControl={
							<RefreshControl onRefresh={refresh} refreshing={loading} />
						}
						showsVerticalScrollIndicator={false}
						style={styles.container}
					>
						<Icon
							color={sto.black}
							name={Platform.OS === 'ios' ? 'ios-print' : 'md-print'}
							size={100}
						/>
						<NoticeView
							buttonText={buttonText}
							header={header}
							onPress={onPress}
							style={styles.notice}
							text={text}
						/>
						{description ? (
							<Text style={styles.description}>{description}</Text>
						) : null}
					</ScrollView>
				)}
			/>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		backgroundColor: sto.white,
	},
	content: {
		flex: 1,
		alignItems: 'center',
		justifyContent: 'center',
	},
	description: {
		marginHorizontal: 30,
		marginVertical: 20,
		textAlign: 'center',
	},
	notice: {
		flex: 0,
	},
})
