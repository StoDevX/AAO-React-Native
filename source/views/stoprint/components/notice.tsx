import * as React from 'react'
import {
	Platform,
	RefreshControl,
	ScrollView,
	StyleSheet,
	Text,
} from 'react-native'
import Icon from 'react-native-vector-icons/Ionicons'
import {NoticeView} from '@frogpond/notice'
import {sto} from '../../../lib/colors'

type Props = {
	buttonText: string
	description?: string
	header: string
	onPress: () => void
	onRefresh?: () => void
	refreshing?: boolean
	text: string
}

export const StoPrintNoticeView = (props: Props): React.ReactElement => {
	let {buttonText, description, header, onPress, text, onRefresh, refreshing} =
		props
	let iconName = Platform.OS === 'ios' ? 'ios-print' : 'md-print'

	return (
		<ScrollView
			contentContainerStyle={styles.content}
			refreshControl={
				onRefresh && refreshing != null ? (
					<RefreshControl onRefresh={onRefresh} refreshing={refreshing} />
				) : undefined
			}
			showsVerticalScrollIndicator={false}
			style={styles.container}
		>
			<Icon color={sto.black} name={iconName} size={100} />
			<NoticeView
				buttonText={buttonText}
				header={header}
				onPress={onPress}
				style={styles.notice}
				text={text}
			/>
			{description ? (
				<Text style={styles.description}>{description}</Text>
			) : null}
		</ScrollView>
	)
}

const styles = StyleSheet.create({
	container: {
		backgroundColor: sto.white,
	},
	content: {
		flex: 1,
		alignItems: 'center',
		justifyContent: 'center',
	},
	description: {
		marginHorizontal: 30,
		marginVertical: 20,
		textAlign: 'center',
	},
	notice: {
		flex: 0,
	},
})
