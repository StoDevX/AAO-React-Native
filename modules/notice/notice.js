// @flow
import * as React from 'react'
import {ActivityIndicator, StyleSheet, Text, View} from 'react-native'
import * as c from '@frogpond/colors'
import {Button} from '../../source/components/button'
import {Heading} from '../../source/components/markdown/heading'
import {Viewport} from '@frogpond/viewport'

const styles = StyleSheet.create({
	container: {
		flex: 1,
		paddingHorizontal: 30,
		justifyContent: 'center',
		alignItems: 'center',
		backgroundColor: c.white,
	},
	text: {
		textAlign: 'center',
	},
	spinner: {
		alignItems: 'center',
		justifyContent: 'center',
		padding: 8,
	},
})

type Props = {
	buttonDisabled?: boolean,
	header?: string,
	text?: string,
	style?: any,
	spinner?: boolean,
	buttonText?: string,
	onPress?: () => any,
	textStyle?: any,
}

export function NoticeView(props: Props) {
	const {header, text, style, textStyle} = props
	const {buttonDisabled, buttonText, onPress} = props
	const {spinner} = props

	return (
		<Viewport
			render={() => (
				<View style={[styles.container, style]}>
					{spinner ? <ActivityIndicator style={styles.spinner} /> : null}

					{header ? (
						<Heading level={1} style={textStyle}>
							{header}
						</Heading>
					) : null}

					<Text selectable={true} style={[styles.text, textStyle]}>
						{text || 'Notice!'}
					</Text>

					{buttonText ? (
						<Button
							disabled={buttonDisabled}
							onPress={onPress}
							title={buttonText}
						/>
					) : null}
				</View>
			)}
		/>
	)
}
