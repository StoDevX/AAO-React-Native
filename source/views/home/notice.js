// @flow

import * as React from 'react'
import * as glamorous from 'glamorous-native'
import * as c from '../components/colors'
import sample from 'lodash/sample'

const messages = [
	'☃️ An Unofficial App Project ☃️',
	'For students, by students',
	'By students, for students',
	'An unofficial St. Olaf app',
	'For Oles, by Oles',
	'☃️',
	'🦁',
]

export function UnofficialAppNotice() {
	return (
		<glamorous.View justifyContent="center" marginHorizontal={10}>
			<glamorous.View backgroundColor="rgba(0,0,0,0.05)" borderRadius={7}>
				<glamorous.Text color={c.black25Percent} padding={8} textAlign="center">
					{sample(messages)}
				</glamorous.Text>
			</glamorous.View>
		</glamorous.View>
	)
}
