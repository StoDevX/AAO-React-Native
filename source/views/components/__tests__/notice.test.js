/* eslint-env jest */
// @flow

import 'react-native'
import * as React from 'react'
import ReactTestRenderer from 'react-test-renderer'

import {NoticeView} from '../notice'

test('renders the given text', () => {
	const tree = ReactTestRenderer.create(<NoticeView text="A Label I Am" />)
	expect(tree).toMatchSnapshot()
})

test('renders a button, if given', () => {
	const tree = ReactTestRenderer.create(
		<NoticeView buttonText="Button" text="Label" />,
	)
	expect(tree).toMatchSnapshot()
})

test('shows an ActivityIndicator if given [spinner]', () => {
	const tree = ReactTestRenderer.create(
		<NoticeView spinner={true} text="Label" />,
	)
	expect(tree).toMatchSnapshot()
})
