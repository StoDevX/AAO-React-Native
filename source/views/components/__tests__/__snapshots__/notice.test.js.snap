// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a button, if given 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 16,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    selectable={true}
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Label
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits="button"
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "rgb(67, 114, 170)",
        "borderRadius": 6,
        "marginVertical": 10,
        "opacity": 1,
        "overflow": "hidden",
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#007aff",
            "fontSize": 17,
            "fontWeight": "500",
            "textAlign": "center",
          },
          null,
          Object {
            "backgroundColor": "transparent",
            "color": "#FFFFFF",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "400",
            "letterSpacing": -0.32,
            "lineHeight": 21,
          },
          null,
        ]
      }
    >
      Button
    </Text>
  </View>
</View>
`;

exports[`renders the given text 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 16,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    selectable={true}
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    A Label I Am
  </Text>
</View>
`;

exports[`shows an ActivityIndicator if given [spinner] 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 16,
      },
      undefined,
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 8,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    selectable={true}
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Label
  </Text>
</View>
`;
